groth16-framework/test_data/Groth16VerifierExtensions.sol

```diff
@@ -44,6 +47,19 @@ contract Query2 is Verifier {
     // The block hash offset in the plonky2 public inputs.
     uint32 constant PI_BLOCK_HASH_OFFSET = PI_NFT_IDS_OFFSET + L * 8;
 
+    // The rewards rate offset in the plonky2 public inputs.
+    uint32 constant PI_REWARDS_RATE_OFFSET = PI_BLOCK_HASH_OFFSET + PACKED_U256_LEN * 8;
+
+    // The ERC20 result offset in the plonky2 public inputs.
+    uint32 constant PI_ERC20_RESULT_OFFSET = PI_REWARDS_RATE_OFFSET + PACKED_U256_LEN * 8;
+
+    // The query identifier offset in the plonky2 public inputs.
+    uint32 constant PI_QUERY_IDENTIFIER_OFFSET = PI_ERC20_RESULT_OFFSET + PACKED_U256_LEN * 8;
+
+    // Supported query identifiers
+    uint256 constant QUERY_IDENTIFIER_NFT = 67;
+    uint256 constant QUERY_IDENTIFIER_ERC20 = 88;
+
     // The query struct used to check with the public inputs.
     struct Query {
```

Comment:
This should give a ~20% gas saving

```suggestion
    struct Query {
        address contractAddress;
        uint96 minBlockNumber;
        address userAddress;
        uint96 maxBlockNumber;
        address clientAddress;
        uint88 rewardsRate;
        uint8 identifier;
        bytes32 blockHash;
    }
```

Reply:
Fixed in commit [d7a6d10](https://github.com/Lagrange-Labs/Euclid-database/pull/21/commits/d7a6d1054fe1853f1d9cf3b8ffa47139efb3792c).